admin/auth.go

```diff
@@ -25,8 +25,54 @@ const (
 )
 
 // Helper to verify if user is an admin
-func checkAdminLevel(level string) bool {
-	return (level == adminLevel)
+func checkAdminLevel(user users.AdminUser) bool {
+	return user.Admin
+}
+
+// Helper to check if query access is granted
+func checkQueryLevel(permissions users.UserPermissions) bool {
+	return permissions.Query
+}
+
+// Helper to check if carve access is granted
+func checkCarveLevel(permissions users.UserPermissions) bool {
+	return permissions.Carve
+}
+
+// Helper to check if environment access is granted
+func checkEnvironmentLevel(permissions users.UserPermissions, environment string) bool {
+	return permissions.Environments[environment]
+}
+
+// Helper to check permissions for a user
+func checkPermissions(username string, query, carve, env bool, environment string) bool {
```

Comment:
this is a little bit funky.  if you do

```
ok := getPermissions("peter", true, true, true, "")
```
what actually gets returned is checkQueryLevel, but nothing on checkCarveLevel or checkEnvironmenLevel.

I'd suggest something like

```
type level int
const (
  queryLevel level = iota
  carveLevel
  envLevel
)

func checkPermissions(username string, permissionsLevel level, environment string) bool {
  switch permissionsLevel {
    case queryLevel:
      ..
    case carveLevel:
    ...
  }
```

that would at least make it more obvious that you're only requesting a single permission.